### ISSUE TITLE
batch user creation not working

### ISSUE THREAD

**Issue by helenvarley** [upstage-org/upstage#1139]
i tried to create a player (in order to test the delete player function) using the batch user creation tool, & got this: 
![Screenshot from 2023-09-28 14-43-09](https://github.com/upstage-org/upstage/assets/3437324/a9f11150-b68b-4548-884f-f351f97ed7ce)

**Comment by teddyfullstack** [comment]
The unique email constraint in #1105 was the cause of this, so I added a new column for entering emails 👇 

<img width="1042" alt="Screenshot 2023-09-30 at 01 27 21" src="https://github.com/upstage-org/upstage/assets/29925961/d6171eac-22fb-4bbf-961b-d499a672226a">


And the way we throw all the error logs into the notification was 🤐 Luckily, we can create new players on Studio now, which got much more elegant error messages 😅 

<img width="920" alt="Screenshot 2023-09-30 at 01 25 32" src="https://github.com/upstage-org/upstage/assets/29925961/d5d83dd4-ca02-46a0-8f15-399d2c0f069f">

**Comment by vickis** [comment]
Thanks for your work there :)
Hhhm have we created an issue for ourselves in wanting to have unique
emails and the ability to clean up the members based on emails double ups
But also needing guest logins for demonstration and walk through events ….

**Comment by helenvarley** [comment]
right - but it will work to have a dummy email address for guest accounts, won't it? 
we can use something like guest1@upstage.live ?

**Comment by helenvarley** [comment]
i just tested this, creating the user guest1 with the dummy email as above, and it worked - but i couldn't see the user in the player list. i tried sorting the list by the date registered column, & nothing happened - the first 3 registered a year ago, then the next one 2 months ago, it is not in any order.
finally i searched for guest1 & found it, & it showed with its status already approved ... is that because it's been made in the Admin section, rather than someone external registering? i guess that's good, as when we create players this way, we know we want them so there doesn't really need to be an approval process.

**Comment by teddyfullstack** [comment]
> i tried sorting the list by the date registered column, & nothing happened

Please make sure you have cancelled sorting on the columns on the left (check the green arrows in the screenshot below), which will take higher priority in multi-sort criteria:

<img width="1338" alt="Screenshot 2023-09-30 at 23 39 07" src="https://github.com/upstage-org/upstage/assets/29925961/a44ed434-330d-4ee2-ae93-f941e007a820">

The player table is not refreshing after new player(s) are created successfully, which is an issue for me to fix 👌

**Comment by teddyfullstack** [comment]
> the multi-sort criteria works like this:

If you enabled sorting on multiple columns, for example:
- Role Descending
- Last Login Ascending
- Date Registered Descending

Then the list would be like this:
- List all Super Admins > then Admins > then Players and Guests
- If there are multiple Admins > They will be sorted by last login
- If multiple Admins with the same last login (which barely exists) > They will be sorted by Date Registered > And this might be the reason why you think it's not working 😅 

In short, if you want to sort only in Date Registered, please cancel sorting on other columns (or I can change the priority/default direction between them on your wish) 👌

**Comment by vickis** [comment]
We can make sort we get the sort priority right rather than more code work
thanks for the detailed explanation! :)

**Comment by helenvarley** [comment]
well today i went to the player management list (having closed & shut down everything yesterday) & without clicking on any other columns i clicked the date registered column to sort the players by the date registered. the result is this:
![Screenshot from 2023-10-04 16:43:59](https://github.com/upstage-org/upstage/assets/3437324/e471fde7-e651-47c7-a698-db025f98bcab)

**Comment by teddyfullstack** [comment]
> without clicking on any other columns

If you check the Role & Last login columns, they are initially sorted, so you will have to cancel it. I can change the default sort column to Date Registered to remove the confusion. What do you think?

**Comment by helenvarley** [comment]
hmm. that is certainly not intuitive - to have to cancel sorting on a column that i did not click to sort, before being able to sort by another column.
however, i tried it now, i cancelled on the role column; when i hovered over the last login column it did not say click to cancel, it said click to sort. then i tried sorting by the date registered and it did not change the order at all. i clicked again, the order still doesn't change.

it's strange that this is so complicated. i remember from the old UpStage we had sortable tables for things & it seemed to work more simply. but if it's not possible to make it simple, would it be better to stick with search & filter tools rather than sortable columns?

**Comment by vickis** [comment]
Agree - Would be great to have a simple search and filter rather than a
complicated system
- especially if this helps from the code end! :)

**Comment by teddyfullstack** [comment]
> it's strange that this is so complicated. i remember from the old UpStage we had sortable tables for things & it seemed to work more simply. but if it's not possible to make it simple, would it be better to stick with search & filter tools rather than sortable columns?

Sorry that it wasn't how you would expect. I was expecting the new multi-sort mechanism would be useful. But let's discuss sorting on another issue. This issue should be focused on batch user creation, and it was resolved 👌
